src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -140,6 +140,16 @@ void emit(CodeWriter codeWriter, String enclosingName, Set<Modifier> implicitMod
     codeWriter.popTypeVariables(typeVariables);
   }
 
+  private CodeBlock javadocWithParameters() {
+    CodeBlock.Builder builder = javadoc.toBuilder();
+    for (ParameterSpec parameterSpec : parameters) {
+      if (!parameterSpec.javadoc.isEmpty()) {
+        builder.add("@param $L $L", parameterSpec.name, parameterSpec.javadoc);
```

Comment:
I'd really like a newline before these tags but it's incredibly annoying to do in Java and 1 line of code in Kotlin. How about
```java
boolean emitTagNewline = true;
for (ParameterSpec parameter : parameters) {
  if (!parameter.javadoc.isEmpty()) {
    if (emitTagNewline) builder.add("\n");
    emitTagNewline = false;
    builder.add("...");
  }
}
```

Reply:
How about

```java
return javadoc.toBuilder().add(
    parameters
        .stream()
        .filter(p -> !p.javadoc.isEmpty())
        .map(p -> CodeBlock.of("@param $L $L", p.name, p.javadoc))
        .collect(CodeBlock.joining("", "\n", "")));
```
It will create (and merge) many intermediate `CodeBlock` but that's probably an acceptable tradeoff (we don't check whether any `ParameterSpec` has a non-empty javadoc either before creating the builder, therefore making an unnecessary copy of the javadoc, so :man_shrugging:)

But how about `@return`, `@throws`, `@see`, etc. (that, in addition, [generally](https://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#orderoftags) come *after* the `@param`s)? If you put them in the method javadoc, there'd be a blank line between them and `@param`s, unless you remember **not** to end your method javadoc with a `\n` in presence of `ParameterSpec`-level javadoc.
Maybe we should add `returnTypeJavadoc` et al. at the same time as `ParameterSpec.javadoc`?

:can_of_worms:

Reply:
That was close to what my Kotlin line was going to be!

I was going to do the other tags in a follow-up PR.

Reply:
Ended up going with the uglier approach. The stream approach would add the extra line even when there were no parameter javadocs since the `collect` was still called.

Reply:
I find the stream to be the uglier approach!